src/ppopt/utils/region_overlap_utils.py

```diff
@@ -0,0 +1,188 @@
+from itertools import permutations
+from typing import List, Tuple, Optional
+from collections import deque
+
+import numpy
+import copy
+
+from ..mpmilp_program import MPMILP_Program
+from ..critical_region import CriticalRegion
+
+from .mpqp_utils import get_bounds_1d
+
+
+def reduce_overlapping_critical_regions_1d(program: MPMILP_Program, regions: List[CriticalRegion]) -> Tuple[
+    List[CriticalRegion], bool]:
+    """
+    This adjusts critical regions for the 1d-parameter case, so that there are no overlaps.
+
+    :param program: the MPMILP_Program to be solved
+    :param regions: a list of CriticalRegion objects
+    :return: a list of CriticalRegion objects and a flag indicating whether there are still overlaps
+    """
+
+    # guard around the statement
+    if program.num_t() != 1:
+        raise ValueError('reduce_overlapping_critical_regions_1d requires a 1d-parameter problem')
+
+    overlaps_remaining, regions = identify_overlaps(program, regions)
+
+    return regions, overlaps_remaining
+
+
+def identify_overlaps(program: MPMILP_Program, regions: List[CriticalRegion]) -> Tuple[bool, list]:
```

Comment:
Needs a docstring and it might help clarity if we change specifically mention that it is for the 1d case